src/django_components/util/types.py

```diff
@@ -1,144 +1,30 @@
-import sys
-import typing
-from typing import Any, Tuple
+from typing import NamedTuple
 
-# See https://peps.python.org/pep-0655/#usage-in-python-3-11
-# NOTE: Pydantic requires typing_extensions.TypedDict until (not incl) 3.12
-if sys.version_info >= (3, 12):
-    from typing import TypedDict
-else:
-    from typing_extensions import TypedDict as TypedDict
 
-try:
-    from typing import Annotated  # type: ignore
-except ImportError:
-
-    @typing.no_type_check
-    class Annotated:  # type: ignore
```

Comment:
No need to redefine the `Annotated` class, since we use the `typing_extensions` package.

And no need to import `TypedDict`, since we now use `NamedTuple` or `dataclass` to define the component typing.